frontend/src/employee-frontend/components/messages/MessagesPage.tsx

```diff
@@ -2,52 +2,96 @@
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
-import ReceiverSelection from 'employee-frontend/components/messages/ReceiverSelection'
-import {
-  deselectAll,
-  getReceiverOptions,
-  SelectorNode
-} from 'employee-frontend/components/messages/SelectorNode'
 import { Loading, Result } from 'lib-common/api'
+import { UUID } from 'lib-common/types'
 import { useRestApi } from 'lib-common/utils/useRestApi'
 import Container from 'lib-components/layout/Container'
 import { Gap } from 'lib-components/white-space'
-import React, { useEffect, useState } from 'react'
+import React, { useContext, useEffect, useState } from 'react'
 import styled from 'styled-components'
-import { getMessagingAccounts } from './api'
+import { deleteDraft, getMessagingAccounts, postMessage } from './api'
 import MessageEditor from './MessageEditor'
 import MessageList from './MessageList'
+import {
+  MessagesPageContext,
+  MessagesPageContextProvider
+} from './MessagesPageContext'
+import ReceiverSelection from './ReceiverSelection'
+import { deselectAll, SelectorNode } from './SelectorNode'
 import Sidebar from './Sidebar'
 import { MessageAccount, MessageBody } from './types'
-import { AccountView } from './types-view'
-import { postMessage } from './api'
-import { UUID } from 'lib-common/types'
 
 const PanelContainer = styled.div`
   display: flex;
 `
 
-export default React.memo(function MessagesPage() {
+function MessagesPage() {
+  const {
+    selectedDraft,
+    setSelectedDraft,
+    view,
+    setView,
+    refreshMessages
+  } = useContext(MessagesPageContext)
+
   const [accounts, setResult] = useState<Result<MessageAccount[]>>(Loading.of())
   const loadAccounts = useRestApi(getMessagingAccounts, setResult)
   useEffect(() => loadAccounts(), [loadAccounts])
```

Comment:
```suggestion
  const [accounts, setResult] = useState<Result<MessageAccount[]>>(Loading.of())
  const loadAccounts = useRestApi(
    getMessagingAccounts,
    (result: Result<MessageAccount[]>) => {
      setResult(result)
      if (result.isSuccess && result.value[0]) {
        setView({
          view: 'RECEIVED',
          account: result.value[0]
        })
      }
    }
  )
  useEffect(() => loadAccounts(), [loadAccounts])
```
One less `useEffect` if you will. Would be even cooler if account handling was fully moved to the context.

Reply:
This would replace the `useEffect` around line 96

Reply:
Moved account handling to context.